op-service/signer/transaction_args.go

```diff
@@ -178,3 +190,41 @@ func (args *TransactionArgs) ToTransactionData() (types.TxData, error) {
 	}
 	return data, nil
 }
+
+type uint256Fields struct {
+	chainID              *uint256.Int
+	maxFeePerGas         *uint256.Int
+	maxPriorityFeePerGas *uint256.Int
+	value                *uint256.Int
+	blobFeeCap           *uint256.Int
+}
```

Comment:
There is `*hexutil.UInt256` you can use in the transaction-args instead. And you can then cast that to a regular uint256 Int like `(*uint256.Int)(myHexutilInt)`. That prevents a lot of error handling and duplicate typing.

Reply:
This is annoying because `DynamicFeeTx` has `big.Int` fields but the other two tx types have `uint256.Int`. So I think we're gonna have to have error handling for the conversions in one of those cases

Reply:
From uint256 to big is guaranteed to not go out of range, and thus doesn't require error handling.

Reply:
`Transaction` is already a safe immutable validated thing, so new `NewTransactionArgsFromTransaction` shouldn't have do checks on the conversion of big-int to uint256 there

Reply:
Made the type changes in https://github.com/ethereum-optimism/optimism/pull/14489/commits/aeccafa532f650ff76de10328c5b00e16e219292